File: hotpot-frontend/src/types/ingredient.ts
// 食材类型定义
export interface Ingredient {
  id?: number;
  name: string;
  code: string;
  categoryId: number;
  unit: string;
  price: number;
  minStock: number;
  maxStock: number;
  supplierId?: number;
  description?: string;
  imageUrl?: string;
  status: number;
  createTime?: string;
  categoryName?: string;
  currentStock?: number;
  stockStatus?: string;
}

export interface IngredientStockVO {
  id: number;
  name: string;  // 注意:后端返回的是name,不是ingredientName
  code: string;
  categoryId: number;
  categoryName: string;
  unit: string;
  price: number | object; // 后端返回BigDecimal类型，可能需要转换
  currentStock: number | object; // 后端返回BigDecimal类型，可能需要转换
  minStock: number | object; // 后端返回BigDecimal类型，可能需要转换
  maxStock: number | object; // 后端返回BigDecimal类型，可能需要转换
  stockStatus: string;  // '正常' | '库存不足' | '库存过剩'
}

export interface UpdateStockParams {
  ingredientId: number;
  type: 'in' | 'out';
  quantity: number;
  operator: string;
  remark?: string;
}

File: src/main/java/com/hotpot/entity/Ingredient.java
// src/main/java/com/hotpot/entity/Ingredient.java
package com.hotpot.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("ingredient")
public class Ingredient {
    @TableId(type = IdType.AUTO)
    private Integer id;

    private String name;
    private String code;
    private Integer categoryId;
    private String unit;
    private BigDecimal price;
    private Integer minStock;
    private Integer maxStock;
    private Integer supplierId;
    private String description;
    private String imageUrl;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    // 关联字段（不映射到数据库）
    @TableField(exist = false)
    private String categoryName;

    @TableField(exist = false)
    private Integer currentStock;

    @TableField(exist = false)
    private String stockStatus;
}